parse/parser_internal_test.go

```diff
@@ -611,3 +611,45 @@ func TestParseInfixExpression(t *testing.T) {
 		}
 	}
 }
+
+func TestParseIfStatement(t *testing.T) {
+	prefixParseFnMap[Bool] = parseBooleanLiteral
+
+	bufs := [][]Token{
+		{
+			{Type: If, Val: "if"},
+			{Type: Lparen, Val: "("},
+			{Type: Bool, Val: "true"},
+			{Type: Rparen, Val: ")"},
+			{Type: Lbrace, Val: "{"},
+			{Type: IntType, Val: "int"},
+			{Type: Ident, Val: "a"},
+			{Type: Rbrace, Val: "}"},
+		},
+	}
+
```

Comment:
i think we should test `else` part also!

Reply:
Yeap. This part was paused. I will do it.